Way of Working

melroy89 opened the pull request:
A way of working proposal written in text. See Files changes tab in this Pull request.

PLEASE, review the agreements. If you do NOT agree let us know now. If you are missing something that is not added, please let us know.

Update: This will become the final version.

TheVillageGuy reviewed (approved):
"Leave IT to the contributor"

e569668 commented on CONTRIBUTING.md:
```diff
@@ -13,7 +13,18 @@ With an account on [GitHub](https://github.com) you will be able to [fork this r
+- Always [create a new Pull Request](https://github.com/MbinOrg/mbin/pulls) in GitHub. Eg. Do _NOT_ merge directly to the `main` branch!
+- All tests should be passing (**all green**) in the PR. This is to avoid regression. More info about the coding style as well as testing see below.
+- Pull Requests require at least one (1) other maintainer approval before the PR gets merged (built-in peer review process).
+- If a contributor has merge right, do _not_ merge his/her PR on their behalf (unless the contibutor agrees you can merge it).
```
suggestion of changing `his/her PR on their behalf` -> `a PR on their behalf`

melroy89 replied:
agreed

melroy89 commented:
> "Leave IT to the contributor"

changed the text.

e569668 reviewed (approved):
seems fine to me. I can imagine the edge case of PRs being left around if the contributor disappears for any reason, but I imagine these are just guidelines and maintainers can agree to take over in those situations where obvious

melroy89 commented:
I also added:

>  Existing GitHub Organization owners are allowed to add another GitHub users to the repository or to the GitHub organization. That being said, _be careful_ who to add and discuss on [Matrix](https://matrix.to/#/#mbin:melroy.org) first.

asdfzdfj commented:
since this this is apparently inspired by zeromq's C4, it may be a good idea to also add a link to the source doc (iirc it's [this one?](https://rfc.zeromq.org/spec/42/)) as a kind of further reading, or as a context for why we have decided to do it this way.

melroy89 commented:
> since this this is apparently inspired by zeromq's C4, it may be a good idea to also add a link to the source doc (iirc it's [this one?](https://rfc.zeromq.org/spec/42/)) as a kind of further reading, or as a context for why we have decided to do it this way.

Agreed, you are completely correct. I got inspiration from C4 by Pieter Hintjens, causing this fork to happen. I included the C4 spec in markdown format as well, extensions or differences from the spec are documented in the guidelines.md file.

TheVillageGuy reviewed (approved):
16 Maintainers MAY merge incorrect patches from other Contributors with the goals of (a) ending fruitless discussions, (b) capturing toxic patches in the historical record, (c) engaging with the Contributor on improving their patch quality.

introducing incorrect patches this way would not seem a good idea to me as it means compromising the working order of the project for the sake of proving someone's point. (If I am interpreting it correctly) The code should always work (*coughs*)

melroy89 commented:
> 16 Maintainers MAY merge incorrect patches from other Contributors with the goals of (a) ending fruitless discussions, (b) capturing toxic patches in the historical record, (c) engaging with the Contributor on improving their patch quality.
> 
> introducing incorrect patches this way would not seem a good idea to me as it means compromising the working order of the project for the sake of proving someone's point. (If I am interpreting it correctly) The code should always work (_coughs_)

I agree, it's a very strange guideline.. Not sure what the rationale behind this one was. Also it's ambitious what they mean with "incorrect". I think the GUIDELINES.md is clear to not cause regression at least with the pipeline.

melroy89 commented:
> Maintainers MAY merge incorrect patches from other Contributors with the goals of (a) ending fruitless discussions, (b) capturing toxic patches in the historical record, (c) engaging with the Contributor on improving their patch quality.

I asked around and asked in the C4 specification matrix room. And the reason is actually simple. If you merge bad code, have a record of proof in git (pull requests aren't forever it's only a github/gitlab thing).

So the idea is if you merge bad code you have proof in the git record that there is a bad actor. You can always revert the commit again or fix it. And the record can act as a proof in case the community want to get rid of bad actors.

TheVillageGuy commented:
> > Maintainers MAY merge incorrect patches from other Contributors with the goals of (a) ending fruitless discussions, (b) capturing toxic patches in the historical record, (c) engaging with the Contributor on improving their patch quality.
> 
> I asked around and asked in the C4 specification matrix room. And the reason is actually simple. If you merge bad code, have a record of proof in git (pull requests aren't forever it's only a github/gitlab thing).
> 
> So the idea is if you merge bad code you have proof in the git record that there is a bad actor. You can always revert the commit again or fix it. And the record can act as a proof in case the community want to get rid of bad actors.

Oh dear, it's filled with my rubbish already :)

BentiGorlich commented:
I agree with almost everything in the PR. 
2 exceptions:
 - I think we should add the guideline to always start with a `WIP` in the title. And remove it when the PR is ready to be merged. This would remove the `don't merge other peoples PR to avoid merging jnfinished code`. I know you said you'd like to merge your own PRs yourself @melroy89 , but I think to make that the general rule doesn't make a lot of sense 😅
 - I think there should be only one document that states whats true. Not one that explains the 'vanilla' C4 approach and then the changes we made.

melroy89 commented:
>     * I think we should add the guideline to always start with a `WIP` in the title. And remove it when the PR is ready to be merged. This would remove the `don't merge other peoples PR to avoid merging jnfinished code`. I know you said you'd like to merge your own PRs yourself @melroy89 , but I think to make that the general rule doesn't make a lot of sense 😅

I still like to merge my own PRs once I think it can be merged, despite having an approval already. But I do not agree with "WIP" since that is from Codeberg 🤣 .. but I do agree of a Draft Pull Request, that is the name that is used on GitHub for WIP PRs. I can add this to the list. But one doesn't necessarily exclude the other, right? We can still do Draft PRs and not merge another maintainer PRs unless with permission.

>     * I think there should be only one document that states whats true. Not one that explains the 'vanilla' C4 approach and then the changes we made.

Well yes, well sort of. Ideally yes 😄 . So we either deviate from C4, and we make changes to that specification. Or we put everything in contributing? Which become way too big. At the same time you noticed we have additional idea or working together (specifically to GitHub or this project), which I would like to write down... but I don't know if we should add all that to the C4? That's why I'm a bit in doubt.
Please let me know if you have a good idea to resolve this doubt I have while keeping the guidelines we find most important digestible and readable?

BentiGorlich commented:
1. Lsts say it this way: I have no problem if other people merge my PRs when they are not draft PRs anymore (didn't know what they're called here). I don't think this should be a general rule, but maybe we just need more opinions on this.
2. I'd put everything in contributing, but add a short summary witt the most important points on top. But I'd say that can be confusing too, so I am not sure...

simonrcodrington commented:
From my perspective, when I complete a proposal and put it up, I'm at a point where I feel the PR resolves the associated issues. I'm happy to have them modified or even merged so long as people are happy with them. 

There's still a chance that we'll merge something in and it'll break, but I'm hoping we'll be about to catch those in the PR or a little later with a bug report.  

So long as we don't let things rot as open PRs I think we'll be good

melroy89 commented:
Anyhow, I extend the contributing.md with the draft PR rule, see example: https://github.com/MbinOrg/mbin/blob/9af8253ae1d2a4cd09b822105c74abe59dd189d3/CONTRIBUTING.md

vintprox reviewed (dismissed):
There are several points I'm stessing about. Hope you all review those as well.

vintprox commented on C4.md:
```diff
@@ -0,0 +1,131 @@
+
+## License
+
+Copyright (c) 2009-2016 Pieter Hintjens. Copyright (c) 2016-2018 The ZeroMQ developers.
```
Right! Just got to know that it's going to be fused with CCONTRIBUTING.md to make a singular document.

melroy89 replied:
I will update the C4.md file with our version of the C4 spec. And... I will link from CONTRIBUTING.md to this C4 file, without any exceptions in CONTRIBUTING, just a link to the C4.md file, which has all the adaptions in line with our (Mbin) WoW.

vintprox commented on C4.md:
```diff
@@ -0,0 +1,131 @@
+### 2.5. Branches and Releases
+
+1. The project SHALL have one branch (“main”, or historically “master”) that always holds the latest in-progress version and SHOULD always build.
+2. The project SHALL NOT use topic branches for any reason. Personal forks MAY use topic branches.
```
I guess, we can just peruse a few statements from C4 that make sense for this project in CONTRIBUTING.md.

melroy89 replied:
When I combine the two guidelines, this rule will be fully removed. Maintainers / owners are now just using topic/feature branches, which is fine IMO. Or even advised.

melroy89 replied:
I rephrased this sentence fully.

melroy89 commented:
My plan now is to indeed merge the C4 and our guidelines together to have one coherent list of rules.

garrettw commented on CONTRIBUTING.md:
```diff
@@ -11,9 +11,32 @@ The code code is mainly written in PHP using the Symfony framework with Twig tem
-Always [create a new Pull Request](https://github.com/MbinOrg/mbin/pulls) in GitHub. Be sure to avoid regression, see below for more info about the coding style as well as testing. All tests in the Pull Request should be green!
+### Way of Working
+
+See below our agreements / guidelines how we work together in the project (in random order):
```
s/how/for how/

garrettw commented on CONTRIBUTING.md:
```diff
@@ -11,9 +11,32 @@ The code code is mainly written in PHP using the Symfony framework with Twig tem
+
+- Respect each other naturally.
+- [Discuss](https://matrix.to/#/#mbin:melroy.org) (big) issues/changes in the community first and have some consensus (WIP: we are still looking into some voting tool that might help here).
+- Always [create a new Pull Request](https://github.com/MbinOrg/mbin/pulls) in GitHub. Thus feature/topic branches are allowed (despite what C4 spec says). Eg. Do _NOT_ merge directly to the `main` branch!
```
If I understand this correctly, the meaning of the last sentence is: "Do _NOT_ merge directly to the main branch without a PR!" Is that right?

melroy89 replied:
Ah yes, I need to clear this up. Of course we can merge into the main branch, but the point was to always use PRs.

melroy89 replied:
I extend the C4 spec to make this all clear.

garrettw commented on CONTRIBUTING.md:
```diff
@@ -11,9 +11,32 @@ The code code is mainly written in PHP using the Symfony framework with Twig tem
+![Tests passing](https://github.com/MbinOrg/mbin/assets/628926/c8cb8778-a60c-49bf-99a2-0770e7148e25)
+
+- Pull Requests require at least one (1) other maintainer approval before the PR gets merged (built-in peer review process).
+- If a contributor has merge right, do _not_ merge a PR on their behalf (unless the contibutor agrees you can merge it).
```
s/right/rights/
s/contibutor/contributor/

garrettw commented on CONTRIBUTING.md:
```diff
@@ -11,9 +11,32 @@ The code code is mainly written in PHP using the Symfony framework with Twig tem
+
+- Pull Requests require at least one (1) other maintainer approval before the PR gets merged (built-in peer review process).
+- If a contributor has merge right, do _not_ merge a PR on their behalf (unless the contibutor agrees you can merge it).
+  Leave it to the contributor when to merge the PR. This is to also avoid regression or merging unfinished work by accident.
```
s/Leave it to the contributor/Let the contributor decide/
s/to also/partly to/
s/merging/merging of/

garrettw commented on CONTRIBUTING.md:
```diff
@@ -11,9 +11,32 @@ The code code is mainly written in PHP using the Symfony framework with Twig tem
+  Leave it to the contributor when to merge the PR. This is to also avoid regression or merging unfinished work by accident.
+  - In case of an _external_ contributor who does not have merge rights, anybody in the community with merge rights can merge the PR.
+- Existing GitHub Organization owners are allowed to add other GitHub users to the repository or to the GitHub organization. That being said, _be careful_ who to add and discuss on [Matrix](https://matrix.to/#/#mbin:melroy.org) first.
+- Do _NOT_ remove GitHub Organization owners based on "inactive for an extended period of time" (despite what C4 specs says). Since we did not yet define extended period of time.
```
s/. Since we did not yet define/, as we have not yet defined the/
Also, it should be either "specs say" or "spec says"

melroy89 replied:
I made it explicit now in C4, around 1 or 2 years inactive.

garrettw commented on CONTRIBUTING.md:
```diff
@@ -11,9 +11,32 @@ The code code is mainly written in PHP using the Symfony framework with Twig tem
+- Do _NOT_ remove GitHub Organization owners based on "inactive for an extended period of time" (despite what C4 specs says). Since we did not yet define extended period of time.
+- Comply with the [Collective Code Construction Contract (C4) - revision 3](C4.md) rules.
+
+The model loosely is based on Collective Code Construction Contract ('C4' for short). For [the complete C4 rules read this spec (rev 3)](C4.md). However, we follow the model but might _deviate_ or _extend_ on this specification (now or in the future). In those cases we deviate or extend; see the bullet list defined above.
```
s/loosely is/is loosely/
s/;/,/

melroy89 replied:
I will merge the C4 spec with our own guidelines, meaning we have one single point of truth. So this whole sentence will be removed.

garrettw reviewed (commented):
I haven't looked over the C4 file yet; I'll do that next.

PaarthShah commented on C4.md:
```diff
@@ -0,0 +1,131 @@
+2.  A patch SHOULD be a minimal and accurate answer to exactly one identified and agreed problem.
+3.  A patch MUST adhere to the code style guidelines of the project if these are defined.
+4.  A patch MUST adhere to the “Evolution of Public Contracts” guidelines defined below.
+5.  A patch SHALL NOT include non-trivial code from other projects unless the Contributor is the original author of that code.
```
I'm a little unsure just by reading this: does this apply to the inclusion of new libraries not written by the contributor? Because if so, I feel like it might be a bit over-restrictive.

We *should* ensure that any libraries introduced don't have negative side effects or licensing issues, but that's a whole other topic.

melroy89 replied:
A "patch" is normally a single problem that need to be fixed. It's not really about libraries, it's about fixing this problem using a patch. The rationale behind this argument is to solve the patch by not introducing non-trivial code from others (assuming it's not causing license issues) but if a patch is needed that is non-trivial the code _could_ only come from the contributor itself. Ideally **all code** should be trivial of course and easy to understand.

PaarthShah replied:
Makes total sense to me; I think I was just over-thinking it and assumed that "patch" was also overloaded to mean "feature implementation", but from the rest of the discussion I think it's truly more limited along the lines of a bugfix.

PaarthShah commented on C4.md:
```diff
@@ -0,0 +1,131 @@
+3.  A patch MUST adhere to the code style guidelines of the project if these are defined.
+4.  A patch MUST adhere to the “Evolution of Public Contracts” guidelines defined below.
+5.  A patch SHALL NOT include non-trivial code from other projects unless the Contributor is the original author of that code.
+6.  A patch MUST compile cleanly and pass project self-tests on at least the principal target platform.
```
```suggestion
6.  A patch MUST compile cleanly and pass project self-tests on at least the principal target platform before a Maintainer can merge it.
```

melroy89 replied:
Thanks. Added

PaarthShah commented on C4.md:
```diff
@@ -0,0 +1,131 @@
+4. The user or Contributor SHOULD seek consensus on the accuracy of their observation, and the value of solving the problem.
+5. Users SHALL NOT log feature requests, ideas, suggestions, or any solutions to problems that are not explicitly documented and provable.
+6. Thus, the release history of the project SHALL be a list of meaningful issues logged and solved.
+7. To work on an issue, a Contributor SHALL fork the project repository and then work on their forked repository.
```
What about a Maintainer who wishes to contribute? Are they expected to use their own fork? Or are they allowed to create a new branch on the main repository? Or is this addressed by point 9?

melroy89 replied:
You need to fork the repo in GitHub, if you do not have the write access in GitHUb. Then you can create a PR from it. Only github org/repo members with write access can also create git branches. This is how GitHub is working.

PaarthShah replied:
Right, that makes sense mechanically, I meant from a rules/expectations perspective; should maintainers also be expected to make PRs based off of their personal forks just like a non-maintainer contributor would be forced to? Or is it acceptable for them to just make a new branch on the main repository itself and make a PR from that?

melroy89 replied:
I updated this section now. TO make it clear how it works in GitHub and how we work.

PaarthShah commented on C4.md:
```diff
@@ -0,0 +1,131 @@
+8. To submit a patch, a Contributor SHALL create a Platform pull request back to the project.
+9. A Contributor SHALL NOT commit changes directly to the project.
+10. If the Platform implements pull requests as issues, a Contributor MAY directly send a pull request without logging a separate issue.
+11. To discuss a patch, people MAY comment on the Platform pull request, on the commit, or elsewhere.
```
I'd go further: discussion for a patch should only be on the PR itself.

Discussion on the *problem* can be on the PR, issue, the matrix channel, et al.

This is to anticipate scenarios where multiple patches are proposed for the same problem.

melroy89 replied:
It should be a "should" indeed. Ir should be a PR and it should be discussed ideally in Matrix, especially when it's a bug fix or any other problem you're trying to fix.

We have a dedicated PR room in Matrix for PRs.

melroy89 replied:
I updated this section in the new version.

PaarthShah commented on C4.md:
```diff
@@ -0,0 +1,131 @@
+10. If the Platform implements pull requests as issues, a Contributor MAY directly send a pull request without logging a separate issue.
+11. To discuss a patch, people MAY comment on the Platform pull request, on the commit, or elsewhere.
+12. To accept or reject a patch, a Maintainer SHALL use the Platform interface.
+13. Maintainers SHOULD NOT merge their own patches except in exceptional cases, such as non-responsiveness from other Maintainers for an extended period (more than 1-2 days).
```
I think there's value in putting a hard deadline to respond here, to remove ambiguity and the "Oh I thought I waited long enough" style of conflict

melroy89 replied:
Updated.

PaarthShah commented on C4.md:
```diff
@@ -0,0 +1,131 @@
+14. Maintainers SHALL NOT make value judgments on correct patches.
+15. Maintainers SHALL merge correct patches from other Contributors rapidly.
+16. Maintainers MAY merge incorrect patches from other Contributors with the goals of (a) ending fruitless discussions, (b) capturing toxic patches in the historical record, (c) engaging with the Contributor on improving their patch quality.
+17. The user who created an issue SHOULD close the issue after checking the patch is successful.
```
The PR should contain "closing keywords" to do this automatically.

melroy89 replied:
Correct. I extend C4 with this example.

PaarthShah commented on C4.md:
```diff
@@ -0,0 +1,131 @@
+15. Maintainers SHALL merge correct patches from other Contributors rapidly.
+16. Maintainers MAY merge incorrect patches from other Contributors with the goals of (a) ending fruitless discussions, (b) capturing toxic patches in the historical record, (c) engaging with the Contributor on improving their patch quality.
+17. The user who created an issue SHOULD close the issue after checking the patch is successful.
+18. Any Contributor who has value judgments on a patch SHOULD express these via their own patches.
```
Clarify: before *and* after it's merged?

melroy89 replied:
After. I updated it.

PaarthShah commented on C4.md:
```diff
@@ -0,0 +1,131 @@
+16. Maintainers MAY merge incorrect patches from other Contributors with the goals of (a) ending fruitless discussions, (b) capturing toxic patches in the historical record, (c) engaging with the Contributor on improving their patch quality.
+17. The user who created an issue SHOULD close the issue after checking the patch is successful.
+18. Any Contributor who has value judgments on a patch SHOULD express these via their own patches.
+19. Maintainers SHOULD close user issues that are left open without action for an uncomfortable period of time.
```
Again, I think it's worth defining an unambiguous amount of time, *but* also allowing Maintainers to keep an issue open longer at their discretion. Stalebots etc can be used to this end.

melroy89 replied:
I remove this sentence, since we now use GitHub Stale action.

melroy89 commented on C4.md:
```diff
@@ -0,0 +1,131 @@
+4.  A patch MUST adhere to the “Evolution of Public Contracts” guidelines defined below.
+5.  A patch SHALL NOT include non-trivial code from other projects unless the Contributor is the original author of that code.
+6.  A patch MUST compile cleanly and pass project self-tests on at least the principal target platform.
+7.  A patch commit message MUST consist of a single short (less than 50 characters) line stating the problem (“Problem: …") being solved, followed by a blank line and then the proposed solution (“Solution: …").
```
I agree. My proposal is to use GitHub PRs for this, explain it in the description text. And use labels where needed.

melroy89 replied:
The title of the PR is key here. since that will be merged together (squashed) when merging the PR.

melroy89 replied:
This section is now extended.

asdfzdfj reviewed (commented):
already mentioned in matrix and sorry for repeating this again but I think this should also be mentioned here:

stumbled upon an interesting remarks regarding this issue/PR/C4 implementation: https://programming.dev/comment/5160514

> It’s great to see the attempt and also an example of what the C4 guidelines are made to avoid.
> 
> Notice how many comments are little nitpicks about this and that. Completely stalling the commit and getting further away from the original point of C4 which is to reduce contributor friction and avoid these kind of endless discussions on PRs.
>
> I don’t want to be too critical because some of that is a clear lack of understanding of the motivations of C4 which is explained more thoroughly in Pieter’s blog posts. You don’t want to adopt a contributor guidelines that you don’t understand of course.
> 
> IMO it’s better just to implement it as-is and start using it in practice rather than bikeshedding.

and this is presumably the blog post in question, from Pieter Hintjens: [Why Optimistic Merging Works Better](http://hintjens.com/blog:106)

-----

I have to admit that after seeing the above remarks, checking out the blog and perhaps the [talk](https://www.youtube.com/watch?v=uzxcILudFWM) mentioned in the blog, I kinda agree with the remark above and that the current discussion here is starting to go into the realm of letting perfect be the enemy of good now.

And so the following review I made is with the intention to finally get this merged and actually used in practice, the refinement that @melroy89 mentioned could come and merged later

LoveIsGrief reviewed (approved):
Agreeing with @asdfzdfj 

This looks good as it is.

asdfzdfj commented:
forgive me but I've taken the liberty to apply wording changes that @garrettw suggested to the CONTRIBUTING.md file, please recheck it again when you have the time.
(actually I have more changes I wanted to add to the guide but don't want to overstep too much at the moment)

also I found [this document](https://zguide.zeromq.org/docs/chapter6/#The-ZeroMQ-Process-C) that goes over the C4 process spec and explain thoughts behind it. highly recommened to give it a read, I think you may find some answers regarding C4 here (at least it did for me).

melroy89 commented on C4.md:
Yup. I'm busy now combining our changes into OUR version of the C4 spec.

melroy89 commented:
As promised I updated C4 spec with our guidelines (I merged it to a single document), that matches our values and way of working now. The CONTRIBUTING.md is already big enough. So I extend/remove/updated the C4 spec accordingly to our needs and WoW and link from the Contributing guide to the C4 spec. 

I also updated the Contributing guide with some non-technical guidelines and hopefully agreements we can all agree on.

Please, review again all. 

Sorry for the delay of this PR (at least everybody had enough time to review!). I contributed first to some other PRs, commits, features, fixes for Mbin. 
I now managed to focus my time and energy to finally finish this PR. You can imagine, this is not just a "single line of code" kind of PR ;).

PaarthShah reviewed (approved):
Overall LGTM! Just minor grammar fixes/clarifying points

melroy89 commented on C4.md:
```diff
@@ -0,0 +1,144 @@
+1. The project SHALL use the share-alike license: AGPL.
+2. All contributions (patches/PRs) to the project source code SHALL use the same license as the project.
+3. All patches / PRs are owned by their authors. There SHALL NOT be any copyright assignment process.
+4. Read the [AGPL license](LICENSE) for more information regarding copyright and licensing.
```
True.

melroy89 commented on C4.md:
```diff
@@ -0,0 +1,144 @@
+4. Administrators MAY remove Maintainers that are long inactive (~1-2 years). Mainly due to security reasons. The Maintainer can always return back, if the person wants to become Maintainer again.
+5. Administrators SHOULD remove Maintainers who repeatedly fail to apply this process accurately.
+6. Administrators SHOULD block or ban “bad actors” who cause stress and pain to others in the project. This should be done after public discussion, with a chance for all parties to speak. A bad actor is someone who repeatedly ignores the rules and culture of the project, who is needlessly argumentative or hostile, or who is offensive, and who is unable to self-correct their behavior when asked to ...
+   If the majority of the existing Maintainers agrees on the removal of the “bad actor” (after giving the “bad actor” time to self-improve), it can then be the final decision.
```
I changed it now slightly. I do mean the currently active maintainers. I think it make sense if the majority of the current active maintainers are either neutral or agrees on the removal of another maintainer, this would then be the final agreement on the decision to proceed with removal.


What do you think?

PaarthShah replied:
Yep that's entirely sensible to me 👌🏾

melroy89 replied:
Thank you for becoming a sponsor! I really appreciate that!! 💜

BentiGorlich reviewed (changes requested):
Other than this small inconsistency I really like this draft, good job melroy :)

melroy89 commented on .github/workflows/stale.yml:
```diff
@@ -5,9 +5,12 @@ on:
   stale:
+    permissions:
+      issues: write
+      pull-requests: write
```
No it won't. PRs will not be closed. Only issues will be closed. The write permission is required to add labels to pull requests like "stale".

BentiGorlich replied:
where is it configured then? :thinking:

melroy89 replied:
https://github.com/MbinOrg/mbin/blob/main/.github/workflows/stale.yml

melroy89 replied:
Ps. this was added 5 months ago already ;)

BentiGorlich replied:
Ah ok :+1:

BentiGorlich commented:
> Other than this small inconsistency I really like this draft, good job melroy :)

Misunderstanding / Misinterpretation by me, sorry :D

melroy89 commented:
As you can see the PR is merged. If you still have feedback you, feel free to comment on this PR or in Matrix or create an issue or a new PR.
